include/deal.II/base/symbolic_function.h

```diff
@@ -0,0 +1,377 @@
+
+
+    /**
+     * Constructor that takes a single string that describes the function
+     * expression as a semicolon separated list of expressions.
+     *
+     * The symbolic expression can use the default argument and the default
+     * symbolic time variable, plus any additional symbols that you may
+     * need, provided that you update the user substitution map that substitutes
+     * all of them before you try to evaluate the function or its derivatives,
+     * by calling update_user_substitution_map().
+     */
+    SymbolicFunction(const std::string &expressions);
+
+    /**
+     * Store and apply the substitution map @p substitutions to each symbolic
+     * component of this Function object.
+     *
+     * Notice that this method will trigger a recomputation of the
+     * gradients, Hessians, and Laplacians of each component.
+     */
+    void
+    update_user_substitution_map(
+      const Differentiation::SD::types::substitution_map &substitutions);
+
+    /**
+     * Return a tensor of position symbols that can be used to define the
+     * expressions of this symbolic function object.
+     *
+     * The default argument is a Tensor<1,dim,Differentiation::SD::Expression>
+     * containing the symbols "x" for `dim` equal to one, "x", "y" for `dim`
+     * equal to two, and "x", "y", "z" for `dim` equal to three.
+     */
+    static Tensor<1, dim, Differentiation::SD::Expression>
+    get_default_symbolic_position();
+
+    /**
+     * Get the actual arguments used for the coordinates in the symbolic
+     * function. This object does not include any user-defined arguments.
+     */
+    const Tensor<1, dim, Differentiation::SD::Expression> &
+    get_symbolic_position() const;
+
+    /**
+     * Get the actual symbolic time in use in this symbolic function.
+     */
+    const Differentiation::SD::Expression &
+    get_symbolic_time() const;
+
+    /**
+     * Get the actual symbolic expressions used in this symbolic function.
+     */
+    const std::vector<Differentiation::SD::Expression> &
+    get_symbolic_function_expressions() const;
+
+    /**
+     * Get the currently stored @p user_substitution_map.
+     */
+    const Differentiation::SD::types::substitution_map &
+    get_user_substitution_map() const;
+
+    /**
+     * Return a SymbolicFunction object that represents the time derivative of
+     * this function. The spatial argument, the symbolic time, and the currently
+     * stored user substitution map are forwarded to the new function.
+     */
+    SymbolicFunction<dim, RangeNumberType>
+    time_derivative() const;
+
+    /**
+     * Return the value of the function at the given point. Unless there is only
+     * one component (i.e., the function is scalar), you should state the
+     * component you want to have evaluated; it defaults to zero, i.e., the
+     * first component.
+     */
+    virtual RangeNumberType
+    value(const Point<dim> &p, const unsigned int component = 0) const override;
```

Comment:
For this and the following few functions, you might as well omit the documentation altogether and just have
```
// documentation inherited from the base class
```